ibridges/irodsconnector/tickets.py

```diff
@@ -43,13 +46,72 @@ def create_ticket(self, obj_path: str, expiry_string: str = '') -> tuple:
             (str, bool)
 
         """
-        ticket = irods.ticket.Ticket(self.sess_man.irods_session)
-        ticket.issue('read', obj_path)
-        logging.info('CREATE TICKET: %s: %s', ticket.ticket, obj_path)
+        ticket = irods.ticket.Ticket(self.session.irods_session)
+        ticket.issue(ticket_type, obj_path)
         expiration_set = False
-        if expiry_string != '':
+        if expiry_string is not None:
             try:
                 expiration_set = ticket.modify('expire', expiry_string) == ticket
             except Exception as error:
-                logging.error('Could not set expiration date: %r', error)
+                self.delete_ticket(ticket)
+                raise Exception('Could not set expiration date: %r', error)
         return ticket.ticket, expiration_set
+
+    @property
+    def all_ticket_strings(self) -> list:
+        return [name for name, _, _, _ in self.all_tickets()]
+
+    def get_ticket(self, ticket_str: str) -> irods.ticket.Ticket:
+        if ticket_str in self.all_ticket_strings:
+            return irods.ticket.Ticket(self.session.irods_session, ticket=ticket_str)
+        else:
+            warnings.warn("Ticket does not exist.")
+        return None
+
+    def delete_ticket(self, ticket: irods.ticket.Ticket):
+        if ticket and ticket.string in self.all_ticket_strings:
+            ticket.delete()
+            self.all_tickets(update=True)
+        else:
+            warnings.warn("Ticket does not exist.")
+
+    def all_tickets(self, update: bool = False) -> list:
+        """retrieves all tickets and their metadata belonging to the user.
+
+        Parameters
+        ----------
+        update : bool
+            Refresh information from server.
+
+        Returns
+        -------
+        list
+            [(ticket string, ticket type, irods obj/coll id, expiry data in epoche)]
+        """
+        user = self.session.username
+        if update or self._all_tickets is None:
+            self._all_tickets = []
+            for row in self.session.irods_session.query(TicketQuery.Ticket).filter(
+                    TicketQuery.Owner.name == user):
+                self._all_tickets.append((row[TicketQuery.Ticket.string],
+                                          row[TicketQuery.Ticket.type],
+                                          self._id_to_path(str(row[TicketQuery.Ticket.object_id])),
+                                          row[TicketQuery.Ticket.expiry_ts]
+                                          ))
+        return self._all_tickets
+
+    def _id_to_path(self, itemid: str) -> str:
+        data_query = self.session.irods_session.query(kw.COLL_NAME, kw.DATA_NAME)
+        data_query = data_query.filter(kw.DATA_ID == itemid)
+
+        if len(list(data_query)) > 0:
+            res = next(data_query.get_results())
```

Comment:
This just takes the first item in the list right? In which cases would there be more than one item? Is that a legitimate case, or is that something we might want to throw an error on?

Reply:
Indeed, this case it should not happen that there are more results since we are trying to map a database identifier to a more human informative value, i.e. the path. I am sorry for the confusion and not having documented the function.